tests/kernel/smp/src/main.c

```diff
@@ -1172,6 +1172,56 @@ ZTEST(smp, test_smp_switch_torture)
 	}
 }
 
+/**
+ * @brief Torture test for cpu affinity code
+ *
+ * @ingroup kernel_smp_tests
+ *
+ * @details Pin thread to a specific cpu. Once thread gets cpu, check
+ *          the cpu id is correct and then thread will give up cpu.
+ */
+#ifdef CONFIG_SCHED_CPU_MASK
+static void check_affinity(void *arg0, void *arg1, void *arg2)
+{
+	ARG_UNUSED(arg1);
+	ARG_UNUSED(arg2);
+
+	int affinity = POINTER_TO_INT(arg0);
+	int counter = 30;
+
+	while (counter) {
```

Comment:
Minor: Please use "counter != 0" for compliance with Rule 14.4 on the coding guidelines.